docs/blog/2020-02-19-how-to-build-multilingual-sites-with-gatsby/index.md

```diff
@@ -0,0 +1,260 @@
+
+This is still configurable in a headless CMS. But you need to think about how your website will render such content. Most content items contain links to other content or include it directly. Do you want to let your visitors know that linked content is available in a different language only, mix the content, or not display it at all?
+
+![Linked content is only available in a different language](./images/link-with-desc.png)
+
+And what if the linked content is available only in a language different from the default one (in this case, English)? A typical example of this is a blog post and its author. When the blog post is in English and the author's bio only in Czech, what will you display?
+
+![Author field is required](./images/required-field.png)
+
+You see, fallbacks are not just about content and its presentation online. It's also about content creation and applied restrictions. Some fields need to be filled, some fields require editors to link an item or two. You need to define what content is allowed and what content is invalid based on the requirements of your client.
+
+![Linked item in free text field](./images/rich-text.png)
+
+## Implementation
+
+The more time you invest in gathering the requirements, the less time you will need to spend on the implementation. So what do we need to implement?
+
+* Localized URLs
+* Identification of user locale
+* Generate language-specific dynamic pages
+* Generate language-specific static pages
+* Update components
+
+To illustrate these steps, I used a sample intranet app that was built on Gatsby and Kentico Kontent and used only a single language - English. You can [find it on GitHub](https://github.com/Simply007/kontent-sample-app-gatsby-intranet).
+
+![Intranet app preview](./images/intranet.png)
+
+The app contains a list of employees and a profile page for every one of them.
+
+Before we jump into code, it's essential to mention plugins. They greatly help with handling multilingual websites on Gatsby, so it would be a waste of time to try and reinvent the wheel here. For the full list, see [Gatsby docs](https://www.gatsbyjs.org/docs/localization-i18n/). I chose the [gatsby-plugin-i18n](https://github.com/angeloocana/gatsby-plugin-i18n) as it is capable of automatic locale identification, handles language-specific static pages, and lets me configure the basics.
+
+```json
+{
+     resolve: `gatsby-plugin-i18n`,
+     options: {
+         langKeyDefault: 'en',
+         langKeyForNull: 'en',
+         prefixDefault: false,
+         useLangKeyLayout: false,
+     },
+ },
+```
+
+This configuration (in `gatsby-config.js`) tells the plugin to use 'en' as the default language code (`langKeyDefault`, `langKeyForNull`) and no prefix (`prefixDefault`). The last option (`useLangKeyLayout`) specifies that the used layout is language invariant.
+
+## Localized URLs
```

Comment:
Seems like this and the following four headers are all one section and should be `h3`s. What do you think?

Reply:
Hi Aisha, wow, thank you so much for the suggested changes! Yes, I will adjust the headers :)